tools/src/test/java/org/apache/kafka/tools/reassign/ReassignPartitionsUnitTest.java

```diff
@@ -0,0 +1,787 @@
+            verifyTopicThrottleResults(results.get(topics.get(1)), "leaderFoo", "");
+        }
+    }
+
+    private void verifyTopicThrottleResults(Config config,
+                                            String expectedLeaderThrottle,
+                                            String expectedFollowerThrottle) {
+        Map<String, String> configs = new HashMap<>();
+        config.entries().forEach(entry -> configs.put(entry.name(), entry.value()));
+        assertEquals(expectedLeaderThrottle,
+            configs.getOrDefault(ReassignPartitionsCommand.topicLevelLeaderThrottle(), ""));
+        assertEquals(expectedFollowerThrottle,
+            configs.getOrDefault(ReassignPartitionsCommand.topicLevelFollowerThrottle(), ""));
+    }
+
+    @Test
+    public void testAlterReplicaLogDirs() {
+        try (MockAdminClient adminClient = new MockAdminClient.Builder().
+            numBrokers(4).
+            brokerLogDirs(Collections.nCopies(4,
+                Arrays.asList("/tmp/kafka-logs0", "/tmp/kafka-logs1"))).
+            build()) {
+
+            addTopics(adminClient);
+
+            Map<TopicPartitionReplica, String> assignment = new HashMap<>();
+
+            assignment.put(new TopicPartitionReplica("foo", 0, 0), "/tmp/kafka-logs1");
+            assignment.put(new TopicPartitionReplica("quux", 1, 0), "/tmp/kafka-logs1");
+
+            assertEquals(
+                set(new TopicPartitionReplica("foo", 0, 0)),
+                alterReplicaLogDirs(adminClient, asScala(assignment))
+            );
+        }
+    }
+
+    public void assertStartsWith(String prefix, String str) {
+        assertTrue(str.startsWith(prefix), String.format("Expected the string to start with %s, but it was %s", prefix, str));
+    }
+
+    @Test
+    public void testPropagateInvalidJsonError() {
+        try (MockAdminClient adminClient = new MockAdminClient.Builder().numBrokers(4).build()) {
+            addTopics(adminClient);
+            assertStartsWith("Unexpected character",
+                assertThrows(AdminOperationException.class, () -> executeAssignment(adminClient, false, "{invalid_json", -1L, -1L, 10000L, Time.SYSTEM)).getMessage());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> scala.collection.immutable.Set<T> set(final T... set) {
+        return mset(set).toSet();
+    }
+
+    @SuppressWarnings({"deprecation", "unchecked"})
+    private <T> scala.collection.mutable.Set<T> mset(final T...set) {
+        return JavaConverters.asScalaSet(new HashSet<>(Arrays.asList(set)));
+    }
+
+    @SuppressWarnings({"deprecation", "unchecked"})
+    private <T> Seq<T> seq(T... seq) {
+        return JavaConverters.asScalaIteratorConverter(Arrays.asList(seq).iterator()).asScala().toSeq();
+    }
+
+    @SuppressWarnings("deprecation")
+    private <K, V> scala.collection.Map<K, V> asScala(Map<K, V> jmap) {
+        return JavaConverters.mapAsScalaMap(jmap);
+    }
```

Comment:
What do you think about using not deprecated converters and make utility methods static.
```suggestion
    @SuppressWarnings("unchecked")
    private <T> scala.collection.immutable.Set<T> set(final T... set) {
        return mset(set).toSet();
    }

    @SuppressWarnings({"deprecation", "unchecked"})
    private <T> scala.collection.mutable.Set<T> mset(final T...set) {
        return JavaConverters.asScalaSet(new HashSet<>(Arrays.asList(set)));
    }

    @SuppressWarnings({"deprecation", "unchecked"})
    private <T> Seq<T> seq(T... seq) {
        return JavaConverters.asScalaIteratorConverter(Arrays.asList(seq).iterator()).asScala().toSeq();
    }

    @SuppressWarnings("deprecation")
    private <K, V> scala.collection.Map<K, V> asScala(Map<K, V> jmap) {
        return JavaConverters.mapAsScalaMap(jmap);
    }
```

Reply:
Apache Kafka still provide compatibility with scala 2.12. 
I doubt your suggestion will work in that mode.

Reply:
Make sense, thanks for clarify.  So, only `static` modifier.